### DISCUSSION TITLE
Drop mobile support?

### DISCUSSION THREAD

**Discussion by golf1052** [signal-csharp/Signal-Windows#214] [General]
[Windows 10 Mobile is no longer supported and will no longer be getting any security updates as of Jan 14, 2020](https://support.microsoft.com/en-au/help/4522812/windows-10-update-kb4522812). The original intention for this project was to support Signal on Windows 10 Mobile but I would like to continue working on it. There's a ton of stuff that's come out in the last year that mobile will not support,

A short list
1. Visual Studio 2019 [deploying](https://docs.microsoft.com/en-us/visualstudio/releases/2019/compatibility#windows-store-and-windows-phone-apps) and [XAML editing](https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.0#universal-windows)
2. [Newer .NET Standard versions](https://github.com/dotnet/standard/blob/master/docs/versions.md). We can only support up to .NET Standard 1.4 right now.
3. [Newer UI stuff](https://docs.microsoft.com/en-us/uwp/toolkits/winui/).
4. [Span\<T\>](https://github.com/dotnet/corert/issues/5725).

I don't use a Windows phone as my daily phone anymore and considering Signal is a security focused product it doesn't make much sense to keep using it. I am however open to keeping Windows 10 Mobile support if there's demand for it.

@Trolldemorted

**Comment by Tourniquet88** [comment]
There is definitively still demand for it. Though compared to other messenger available on the platform, the signal app is still very basic and isn't in its current state to be a considerable replacement for messenger like WhatsApp, Threema or Telegram. 

Also regarding the UI stuff, as far as we learned from Build, WinUI 3 should be fully supported by 15063 when its final version arrives, so in terms of UI stuff, Mobile shouldn't be a problem.

The biggest problem with VS 2019 is probably debugging and of course newer .NET standard versions, but are they really necessary?

Regarding security, I would still argue that Windows 10 Mobile is "more secure" even without its security updates, because of the low usage numbers alone. Even iOS is way less targeted for maleware attacks and stuff than Android, only because Android is way much more used. So in that sense, Windows 10 Mobile should be still *secure*.

**Comment by Trolldemorted** [comment]
Sorry for the long delay, its nice to hear from you! Newer .net standard versions would be nice indeed, because our dependencies (e.g. ef core) might stop working/receiving updates for 1.4.
Is building xaml files for W10M-compatible apps no longer possible with VS 2019?

I have to admit I am fairly out of the loop right now and have never heard of WinUI before. Will it enable us to maintain a single codebase for W10M and W10? Warnings like [this one](https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/1946#issuecomment-398898605) made me fear that we might have to split codebases in the near future. Also I heard stories about parts of the windows store shutting down for mobile devices, will/is Signal-Windows affected by that?

Unfortunately we haven't updated libsignal in a while, and we have to tackle that soon-ish. I don't know how long the signal server will accept messages without sealed senders, so we should switch to them before it suddenly stops.

**Comment by Tourniquet88** [comment]
It is still very much possible to build Windows 10 Mobile compatible apps with VS 2019, but you need to target at least 16299 to be able to use the XAML designer as far as I know. 

The biggest problem with using VS 2019 though is that you can't deploy and can't debug anymore.

The Windows Phone 7 - 8.1 Store was shutdown in December, the store on Windows 10 Mobile is still there and working. Even Microsoft Office is still supporting phones till next year.

WinUI is a controls library just like the community toolkit but with the focus on fluent design. So with the current version 2.3 you see a lot of new styling in there which looks a lot more like the Android UI / Windows 10X UI in many cases.
To use version 2.3 you need 15063 as the minimum version which is already the case so no problem there.

https://docs.microsoft.com/en-us/uwp/toolkits/winui/release-notes/
https://www.microsoft.com/de-de/p/xaml-controls-gallery/9msvh128x2zt

**Comment by Trolldemorted** [comment]
Not being able to debug sounds ugly :(

@golf1052 did you have a look at the upcoming [group system](https://signal.org/blog/signal-private-group-system/)? It looks like a lot of working is coming, and unfortunately I am super busy with work and university this semester.

**Comment by ahorn42** [comment]
I would really love to see further support for Windows 10 Mobile, as of now, I think Telegram (with Unigram Mobile App) and your app are the last working (kind of trustworthy and spread) messaging apps left for this platform.

And at least speaking for me I plan to use my Lumia 950XL as long as possible as I still couldn't yet find a suitable alternative without going to Android or iOS, which I both really don't like and most other OSs only work on low-end hardware. But with the lack of messaging apps the phone get's even more annoying as with the limitations that came with Edge as the only available browser, while every thing else still works very well for me. The Hardware is also in a still perfect condition after nearly 5 years now (only replaced the battery once) and I would really like to avoid unnecessary waste of resources because of lacking software, too hard to see that all major companies doing this, but with Open Source we have the chance to not do this :)

Anyway, if you might plan at some point to drop the support, please be so kind to add a tag to the last working release/commit so it would be easier to fork it ;)
